src/main/File/Services/UploadFileService.java

```diff
@@ -0,0 +1,256 @@
+package File.Services;
+
+import Config.Message;
+import Config.Service;
+import File.FileMessage;
+import File.FileType;
+import PDF.PdfMessage;
+import Security.EncryptionController;
+import User.UserMessage;
+import User.UserType;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.gridfs.GridFSBucket;
+import com.mongodb.client.gridfs.GridFSBuckets;
+import com.mongodb.client.gridfs.model.GridFSFile;
+import com.mongodb.client.gridfs.model.GridFSUploadOptions;
+import com.mongodb.client.model.Filters;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.interactive.digitalsignature.PDSignature;
+import org.apache.pdfbox.pdmodel.interactive.digitalsignature.SignatureOptions;
+import org.apache.pdfbox.pdmodel.interactive.digitalsignature.visible.PDVisibleSigProperties;
+import org.apache.pdfbox.pdmodel.interactive.digitalsignature.visible.PDVisibleSignDesigner;
+import org.apache.pdfbox.pdmodel.interactive.form.PDField;
+import org.apache.pdfbox.pdmodel.interactive.form.PDNonTerminalField;
+import org.apache.pdfbox.pdmodel.interactive.form.PDSignatureField;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.bson.types.ObjectId;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.time.LocalDate;
+import java.util.Calendar;
+import java.util.LinkedList;
+import java.util.List;
+
+import static com.mongodb.client.model.Filters.eq;
+
+@Slf4j
+public class UploadFileService implements Service {
+  public static final int CHUNK_SIZE_BYTES = 100000;
+
+  String uploader;
+  String organizationName;
+  UserType privilegeLevel;
+  String filename;
+  String title;
+  String fileContentType;
+  String fileIdStr;
+  boolean annotated;
+  boolean toSign;
+  InputStream fileStream;
+  InputStream signatureFileStream;
+  FileType fileType;
+  MongoDatabase db;
```

Comment:
Why do we still have the MongoDatabase in this serrvice? Why can't we just move the implementation of saving the file to the filedao.save() function here? We should only include the fileDao in these services and let the dao handle the logic

Reply:
its still there?